src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -150,17 +158,19 @@ private class CacheItem
         /// <param name="dateTimeProvider">Provider of time functions.</param>
         /// <param name="loggerFactory">Factory to be used to create logger for the puller.</param>
         /// <param name="stakeChainStore">Storage of POS block information.</param>
-        private CachedCoinView(IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory, INodeStats nodeStats, StakeChainStore stakeChainStore = null)
+        /// <param name="rewindDataIndexStore">Rewind data index store.</param>
+        private CachedCoinView(IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory, INodeStats nodeStats, StakeChainStore stakeChainStore = null, IRewindDataIndexStore rewindDataIndexStore = null)
         {
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.dateTimeProvider = dateTimeProvider;
             this.stakeChainStore = stakeChainStore;
+            this.rewindDataIndexStore = rewindDataIndexStore;
             this.MaxItems = CacheMaxItemsDefault;
             this.lockobj = new AsyncLock();
             this.cachedUtxoItems = new Dictionary<uint256, CacheItem>();
             this.performanceCounter = new CachePerformanceCounter(this.dateTimeProvider);
             this.lastCacheFlushTime = this.dateTimeProvider.GetUtcNow();
-            this.cachedRewindDataList = new List<RewindData>();
+            this.cachedRewindDataList = new List<(int height, RewindData rewindData)>();
```

Comment:
Why not a dictionary then? we will always have only one height ever stored.

Reply:
I wanted to keep changes in CachedCoinView to a minimum as changing it to dictionary changes the logic in CoinView (searching, getting last item, etc.). I have changed it to a SortedDictionary for now to keep it as close to the original logic as possible.